atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/SweepStrategy.java

```diff
@@ -29,27 +29,35 @@
 public final class SweepStrategy {
 
     public static final SweepStrategy CONSERVATIVE =
-            new SweepStrategy(Optional.of(SweeperStrategy.CONSERVATIVE), false);
-    public static final SweepStrategy THOROUGH = new SweepStrategy(Optional.of(SweeperStrategy.THOROUGH), true);
+            new SweepStrategy(Optional.of(SweeperStrategy.CONSERVATIVE), false, false);
+    public static final SweepStrategy THOROUGH = new SweepStrategy(Optional.of(SweeperStrategy.THOROUGH), false, true);
 
     private final Optional<SweeperStrategy> sweeperStrategy;
-    private final boolean mustCheckImmutableLockAfterReads;
+    private final boolean mustCheckImmutableLockAfterNonEmptyReads;
+    private final boolean mustCheckImmutableLockAfterEmptyReads;
```

Comment:
should we split these as:
- mustCheckImmutableLockIfAllCellsReadAndPresent
- mustCheckImmutableLockIfEmptyCellWasPossiblyRead

to make it clear when we need to handle the 'empty reads' case?